utilities/rpp-unittests/OCL/Single_ocl.cpp

```diff
@@ -0,0 +1,533 @@
+    Rpp32f coordinates[4] = {100, 200, 200, 400};
+    Rpp32f hueShift = 10;
+    Rpp32f saturationFactor = 10;
+    Rpp8u minThreshold = 10;
+    Rpp8u maxThreshold = 30;
+    Rpp32u numOfPixels = 4;
+    Rpp32u gaussianKernelSize = 7;
+    Rpp32f kValue = 1;
+    Rpp32f threshold = 15;
+    Rpp32u nonmaxKernelSize = 5;
+    Rpp32u sobelType = 1;
+    Rpp8u min = 10;
+    Rpp8u max = 30;
+    Rpp32u crop_pos_x = 100;
+    Rpp32u crop_pos_y = 100;
+    Rpp32u xRoiBegin = 50;
+    Rpp32u yRoiBegin = 50;
+    Rpp32u xRoiEnd = 200;
+    Rpp32u yRoiEnd = 200;
+    Rpp32u mirrorFlag = 0;
+
+    Rpp32f perspective[9];
+
+    perspective[0] = 1;
+    perspective[1] = 0;
+    perspective[2] = 0.5;
+    perspective[3] = 0;
+    perspective[4] = 1;
+    perspective[5] = 0.5;
+    perspective[6] = 1;
+    perspective[7] = 0;
+    perspective[8] = 0.5;
+    Rpp32f percentage = 100;
+    Rpp32u numbeoOfShadows = 12;
+    Rpp32u maxSizeX = 12;
+    Rpp32u maxSizey = 15;
+    Rpp32u extractChannelNumber = 1;
+
+    switch (test_case)
+    {
+    case 0:
+        test_case_name = "contrast";
+        rppi_contrast_u8_pkd3_gpu(d_input, srcSize[0], d_output, newMin, newMax, handle);
+        break;
+    case 1:
+        test_case_name = "jitter";
+        rppi_jitter_u8_pkd3_gpu(d_input, srcSize[0], d_output, kernelSize, handle);
+        break;
+    case 2:
+        test_case_name = "blur";
+        rppi_blur_u8_pkd3_gpu(d_input, srcSize[0], d_output, kernelSize, handle);
+        break;
+    case 3:
+        test_case_name = "brightness";
+        rppi_brightness_u8_pkd3_gpu(d_input, srcSize[0], d_output, alpha, beta, handle);
+        break;
+    case 4:
+        test_case_name = "blend";
+        rppi_blend_u8_pkd3_gpu(d_input, d_input_second, srcSize[0], d_output, alpha, handle);
+        break;
+    case 5:
+        test_case_name = "color_temperature";
+        rppi_color_temperature_u8_pkd3_gpu(d_input, srcSize[0], d_output, adjustmentValue, handle);
+        break;
+    case 6:
+        test_case_name = "gamma_correction";
+        rppi_gamma_correction_u8_pkd3_gpu(d_input, srcSize[0], d_output, gamma, handle);
+        break;
+    case 7:
+        test_case_name = "fog";
+        rppi_fog_u8_pkd3_gpu(d_input, srcSize[0], d_output, fogValue, handle);
+        break;
+    case 8:
+        test_case_name = "snow";
+        rppi_snow_u8_pkd3_gpu(d_input, srcSize[0], d_output, snowPercentage, handle);
+        break;
+    case 9:
+        test_case_name = "lens_correction";
+        rppi_lens_correction_u8_pkd3_gpu(d_input, srcSize[0], d_output, strength, zoom, handle);
+        break;
+    case 10:
+        test_case_name = "noise";
+        rppi_noise_u8_pkd3_gpu(d_input, srcSize[0], d_output, noiseProbability, handle);
+        break;
+    case 11:
+        test_case_name = "pixelate";
+        rppi_pixelate_u8_pkd3_gpu(d_input, srcSize[0], d_output, handle);
+        break;
+    case 12:
+        test_case_name = "exposure";
```

Comment:
Codacy found an issue: [Variable 'test_case_name' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)